src/com/palmergames/bukkit/towny/utils/PlayerCacheUtil.java

```diff
@@ -377,6 +378,28 @@ else if (CombatUtil.isEnemy(resident.getTown(), town)) {
 	 * @return true if allowed.
 	 */
 	private static boolean getPermission(Player player, TownBlockStatus status, WorldCoord pos, Material material, TownyPermission.ActionType action) {
+		// Allow admins to have ALL permissions
+		TownyUniverse townyUniverse = TownyUniverse.getInstance();
+		if (townyUniverse.getPermissionSource().isTownyAdmin(player))
+			return true;
+
+		//If town is bankrupt, nobody can build
+		TownBlock townBlock = null;
+		Town targetTown = null;
+		if(TownySettings.isTownBankruptcyEnabled() && action == ActionType.BUILD) {
+			try {
+				townBlock = pos.getTownBlock();
+				targetTown = townBlock.getTown();
+				if(targetTown.getAccount().isBankrupt())  {
+					cacheBlockErrMsg(player, TownySettings.getLangString("msg_err_bankrupt_town_cannot_build"));
+					return false;
+				}
+			} catch (NotRegisteredException e) {
+				//ignored
+			} catch (EconomyException ee) {
+				//also ignored
+			}
+		}

```

Comment:
I remember you said there was a reason this had to go here, but I cannot remember what it was. Can you remind me what went wrong when it was done differently?